rasa/core/evaluation/marker_base.py

```diff
@@ -308,6 +309,8 @@ def evaluate_events(
 
         extracted_markers: List[Dict[Text, List[EventMetaData]]] = []
         for session, start_idx in sessions_and_start_indices:
+            if len(session) == 0:  # skip trailing empty sessions
+                continue
```

Comment:
```suggestion
            if len(session) <= 1:  # skip trailing empty sessions
                continue
```

The check with `len(session)==0` does not solve the problem described in the issue: The `_split_sessions` method partitions the given list of events such that each partition starts with an `ActionExecuted` event with the name `ACTION_SESSION_START_NAME`. 

Hence, the `session` list here will be empty if and only if the tracker contains no events at all -- and it contains exactly 1 element if and only if the session only consist of the `ActionExecuted` described before.

By checking for `<=1` we'll cover both cases, i.e. that the tracker is completely empty (which probably cannot even happen, but doesn't hurt) and that the session is effectively empty.

Reply:
Guess it would be good to also add a test for this :)

Reply:
Sure I'll add a test 👍 
I tested it manually and it works. For some reason that session with the one event that I described becomes a session with zero events. I'll look into it to find out why + add a test.

Reply:
1. No, the above cannot work -- Please read the comment above carefully and let me know if you have any questions
2.  I overlooked the "followed by session_started: null."  => Means even the approach with "<=1" won't work properly since sessions containing only the `ActionExecuted` with `ACTION_SESSION_START_NAME` name and that session restart event will contain those 2 events. => We'll have to check what is in the list of events called `session` in the code snippet above.

Reply:
> No, the above cannot work

Sorry, let me clarify. In the original issue I linked an example input and output. For that example, adding `if len(session) == 0:` gave me the output that I expect.

My original description of the issue was a guess about the problem. But after debugging further I realised that `split_sessions` is returning one empty session, then another session with remaining events. I added a test to illustrate the problem. The PR is not ready to re-review since I haven't fixed the problem, but the test should help illustrate the issue. I'll keep looking into it. 

Sorry this won't be fixed for RC2, but I'll fix it for RC3.